shared/constants/metametrics.ts

```diff
@@ -0,0 +1,637 @@
+  /**
+   * A number representing the amount of all NFTs the user possesses across all
+   * networks and accounts.
+   */
+  number_of_nfts?: number;
+  /**
+   * The total number of token contracts the user has across all networks and
+   * accounts.
+   */
+  number_of_tokens?: number;
+  /**
+   * Does the user have the OpenSea API enabled?
+   */
+  opensea_api_enabled?: boolean;
+  /**
+   * Does the user have 3Box sync enabled?
+   *
+   * @deprecated
+   */
+  three_box_enabled?: boolean;
+  /**
+   * Which theme the user has selected.
+   */
+  theme?: string;
+  /**
+   * Does the user have token detection enabled?
+   */
+  token_detection_enabled?: boolean;
+  /**
+   * Does the user have desktop enabled?
+   */
+  desktop_enabled?: boolean;
+  /**
+   * Whether the security provider feature has been enabled.
+   */
+  security_providers?: string[];
+};
+
+export enum MetaMetricsUserTrait {
+  /**
+   * Identified when the user adds or modifies addresses in the address book.
+   */
+  AddressBookEntries = 'address_book_entries',
+  /**
+   * TODO
+   */
+  InstallDateExt = 'install_date_ext',
+  /**
+   * Identified when the Ledger Live connection type is changed.
+   */
+  LedgerConnectionType = 'ledger_connection_type',
+  /**
+   * Identified when the user modifies networks.
+   */
+  NetworksAdded = 'networks_added',
+  /**
+   * Identified when the user modifies networks that lack a ticker.
+   */
+  NetworksWithoutTicker = 'networks_without_ticker',
+  /**
+   * Identified when the "Autodetect NFTs" feature is toggled.
+   */
+  NftAutodetectionEnabled = 'nft_autodetection_enabled',
+  /**
+   * Identified when identities change.
+   */
+  NumberOfAccounts = 'number_of_accounts',
+  /**
+   * The number of unique NFT addresses.
+   */
+  NumberOfNftCollections = 'number_of_nft_collections',
+  /**
+   * TODO
+   */
+  NumberOfNfts = 'number_of_nfts',
+  /**
+   * Identified when the number of tokens change.
+   */
+  NumberOfTokens = 'number_of_tokens',
+  /**
+   * Identified when the OpenSea API is enabled.
+   */
+  OpenSeaApiEnabled = 'opensea_api_enabled',
+  /**
+   * TODO
+   */
+  Theme = 'theme',
+  /**
+   * Identified when the 3Box feature is toggled.
+   *
+   * @deprecated
+   */
+  ThreeBoxEnabled = 'three_box_enabled',
+  /**
+   * Identified when the token detection feature is toggled.
+   */
+  TokenDetectionEnabled = 'token_detection_enabled',
+  /**
+   * Identified when the user enables desktop.
+   */
+  DesktopEnabled = 'desktop_enabled',
+  /**
+   * Identified when the security provider feature is enabled.
+   */
+  SecurityProviders = 'security_providers',
+}
+
+/**
+ * Mixpanel converts the zero address value to a truly anonymous event, which
+ * speeds up reporting
+ */
+export const METAMETRICS_ANONYMOUS_ID = '0x0000000000000000';
+
+/**
+ * Used to identify events that are triggered by the background process.
+ */
+export const METAMETRICS_BACKGROUND_PAGE_OBJECT: MetaMetricsPageObject = {
```

Comment:
Once we upgrade TypeScript we should be able to use `satisfies MetaMetricsPageObject` instead of typing it like this, but this should work for the time being.